### PR TITLE
fix: jump to 404 when be removed from page

### LINKED ISSUE #1079: Unable to redirect on exiting workspace

### PR DESCRIPTION
**Pull request by himself65**
Fixes: https://github.com/toeverything/AFFiNE/issues/1079, https://github.com/toeverything/AFFiNE/issues/1081

## Known issue

Remove event will be delayed for about 10s (backend issue) /cc @zuoxiaodong0815

### CODE REVIEW

File: apps/web/src/pages/workspace/[workspaceId]/[pageId].tsx [hunk 1]
```diff
@@ -124,6 +125,16 @@ const PageDefender = ({ children }: PropsWithChildren) => {
...
+  const api = useGlobalStateApi();
+  useEffect(
+    () =>
+      dataCenter.onWorkspacesChange(({ deleted }) => {
+        if (deleted?.some(workspace => workspace.id === currentWorkspace?.id)) {
+          router.replace('/404?code=kicked');
+        }
+      }),
+    [api, currentWorkspace?.id, dataCenter, router]
+  );
```
Comment (pengx17) [inline apps/web/src/pages/workspace/[workspaceId]/[pageId].tsx:137]: There is some similar 404 handling in `useEnsureWorkspace`. Do you think it is better handled there?

Comment (himself65) [reply apps/web/src/pages/workspace/[workspaceId]/[pageId].tsx:137]: We could do this later, because this is urgent bug need to be fixed

**Comment by thorseraq** [comment]
Also fixes https://github.com/toeverything/AFFiNE/issues/1081

**Review by pengx17 — APPROVED** [review]

**Comment by himself65** [comment]
housekeeping needed, see https://github.com/toeverything/AFFiNE/pull/1090#discussion_r1109351611

**Pull request merged**